rclcpp/test/rclcpp/test_client.cpp

```diff
@@ -327,6 +327,26 @@ TEST_F(TestClientWithServer, test_client_remove_pending_request) {
   EXPECT_TRUE(client->remove_pending_request(future));
 }
 
+TEST_F(TestClientWithServer, prune_requests_older_than_no_pruned) {
+  auto client = node->create_client<test_msgs::srv::Empty>(service_name);
+  auto request = std::make_shared<test_msgs::srv::Empty::Request>();
+  auto future = client->async_send_request(request);
+  auto time = std::chrono::system_clock::now() + 1s;
+
+  EXPECT_EQ(size_t(1), client->prune_requests_older_than(time));
```

Comment:
I would rather use:

```suggestion
  EXPECT_EQ(1u, client->prune_requests_older_than(time));
```

or maybe `size_t{1}`.